script.service.ltvpl/resources/lib/Data/PL_DataSet.py

```diff
@@ -0,0 +1,684 @@
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with XBMC; see the file COPYING.  If not, write to
+#  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+#  http://www.gnu.org/copyleft/gpl.html
+#
+import json
+import time
+from datetime import datetime, timedelta
+from resources.lib.Network.myPickle_io import myPickle_io
+from resources.lib.Network.myJson_io import myJson_io
+from resources.lib.Network.utilities import genericEncode, encodeError
+from resources.lib.Utilities.DebugPrint import DbgPrint
+from resources.lib.Utilities.Messaging import NotificationAction, OpStatus
+from resources.lib.Utilities.PythonEvent import Event
+from .PlayListItem import PlayListItem, isPlayListItem, RecurrenceOptions, ALARMPADDING
+from .fileManager import fileManager, FileManagerMode
+from resources.lib.KodiLib.KodiUtilities import kodiObj, changeChannelByChannelNumber, playerStop, getBroadcast_startTimeList, KODI_ENV
+from resources.lib.Utilities.Messaging import Cmd, VACATIONMODE, PREROLLTIME
+from resources.lib.Utilities.AlarmsMgr import _Alarms
+from resources.lib.Utilities.VirtualEvents import TS_decorator
+if KODI_ENV:
+    import xbmcgui
+    import xbmc
+
+__Version__ = "1.1.1"
+
+DELETE_WAIT_TIME=1
+MODULEDEBUGMODE=True
+LTVPL = 'Live TV Playlist'
+#DbgPrint=LoggerSetup("PL_DataSet")
+
+        
+class PL_DataSet(list,myPickle_io,myJson_io):
+    """
+        Events:
+            ItemRemovedEvent -> Fired when a PlayList item is removed from the dataset
+            ItemCancelledEvent -> Fired when a PlayList item is cancelled
+    """
+    def __init__(self, dataFileName, vacationmode=False, autocleanMode=True):
+        list.__init__(self)
+        myPickle_io.__init__(self)
+        self.ItemRemovedEvent=Event("ItemRemovedEvent",spawn=True)
+        self.ItemCancelledEvent=Event("ItemCancelledEvent",spawn=True)
+        self.ItemAddedEvent=Event("ItemAddedEvent",spawn=True)
+        self.ItemUpdatedEvent=Event("ItemUpdatedEvent",spawn=True)
+        self.SettingsChangedEvent=Event("SettingsChangedEvent", spawn=True)
+        self.skipOperationActive = False
+        self.vacationmode = vacationmode
+        self.autocleanmode = autocleanMode
+        self.prerolltime = ALARMPADDING
+        self.abortOperation = False
+        self.lastChannel = None
+        self.fileManager=fileManager(self, dataFileName, mode=FileManagerMode.PICKLE)
+        DbgPrint("***Calling Data Restore Operation...")
+        self.fileManager.restore()
+        DbgPrint("***LastChannel: {}".format(self.lastChannel))
+        if KODI_ENV and self.lastChannel is None:
+            DbgPrint("*****LastChannel->: {} isNone:{}".format(self.lastChannel, self.lastChannel is None))
+            xbmcgui.Dialog().notification(LTVPL, "Scanning Channel List...")
```

Comment:
string needs to be localized as it is presented to the user

Reply:
done!